lib/galaxy/datatypes/anvio.py

```diff
@@ -5,11 +5,18 @@
 import glob
 import logging
 import os
-from typing import Optional
+from typing import (
+    Optional,
+    TYPE_CHECKING,
+)
 
+from galaxy.datatypes.data import GeneratePrimaryFileDataset
 from galaxy.datatypes.metadata import MetadataElement
 from galaxy.datatypes.text import Html
 
+if TYPE_CHECKING:
+    from galaxy.model import DatasetInstance
```

Comment:
A Protocol might be a good idea here, I guess this only needs to provide `file_name` ?

Reply:
Even if more is needed than file_name in a lot of cases - tracking what is needed would be great. I would love to see a second application to these datatype modules decoupled from our models and Galaxy - there is a lot of useful code and domain knowledge in here and while I've never been able to strictly separate it from the models (hence galaxy-data packaging them both) the actual interconnectedness seems like it could be pretty loose.

Reply:
On decoupling from the model. We import a lot from `model.metadata` in `datatypes/metadata.py`. But other than that, it's just DatasetInstance, HDA, and one case of HDCA. Setting up Protocols for those might be not too difficult. I might try it during next week's "typing-a-thon".